registration_system.cpp

```diff
@@ -0,0 +1,23 @@
+/*
+    Author: Alexis Osnaya.
+    URL: https://codeforces.com/problemset/problem/4/C
+*/
+
+#include <bits/stdc++.h>
+
+using namespace std;
+
+int main(){
+    int cases;
+    string str;
+    map<string,int> mp;
+    map<string,int> :: iterator it;
```

Comment:
No es buena practica que declares variables afuera y no sean utilizadas, siempre tienen que ir al mismo nivel que las usas, en este caso esta variable debe de ir adentro del for.